modules/auxiliary/admin/postgres/postgres_readfile.rb

```diff
@@ -17,25 +18,21 @@ def initialize(info = {})
           as well as read privileges to the target file.
       },
       'Author'         => [ 'todb' ],
-      'License'        => MSF_LICENSE
+      'License'        => MSF_LICENSE,
+      'SessionTypes' => %w[PostgreSQL]
     ))
 
     register_options(
       [
+        OptString.new('DATABASE', [ false, 'The database to authenticate against', 'template1']),
+        OptString.new('USERNAME', [ false, 'The username to authenticate as', 'postgres']),
```

Comment:
oof kinda sucks to have to add this into every postgres session compatible module
I fixed a similar issue with the smb session stuff here https://github.com/rapid7/metasploit-framework/blob/3c16ec2ad896f8ff6fc9fc68111f3617048e6d73/lib/msf/core/optional_session.rb#L13-L22

it's not ideal though maybe better than copy pasta? 🤷